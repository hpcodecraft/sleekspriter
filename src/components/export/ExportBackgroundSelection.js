import React from "react";
import PropTypes from "prop-types";
import { ColorswatchPicker } from "../common";
import { t } from "../../utils";

const ExportBackgroundSelection = ({ background, setBackground }) => {
  return (
    <div>
      <h6>{t("Background")}</h6>
      <ul>
        <li>
          <ColorswatchPicker color={background} action={setBackground} />
        </li>
      </ul>
    </div>
  );
};

ExportBackgroundSelection.propTypes = {
  background: PropTypes.string.isRequired,
  setBackground: PropTypes.func.isRequired,
};

export default ExportBackgroundSelection;
