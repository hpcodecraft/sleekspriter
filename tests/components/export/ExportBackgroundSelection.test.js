import ExportBackgroundSelection from "components/export/ExportBackgroundSelection";

describe("ExportBackgroundSelection", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      background: "#cccccc",
      setBackground: jest.fn(),
    };
    wrapper = shallow(<ExportBackgroundSelection {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
