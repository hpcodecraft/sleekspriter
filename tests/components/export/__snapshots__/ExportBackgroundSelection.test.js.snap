// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportBackgroundSelection should render correctly 1`] = `"<div><h6>Background</h6><ul><li><div><div class=\\"colorswatch\\" style=\\"background:#cccccc\\" title=\\"#cccccc\\"></div></div></li></ul></div>"`;
